app/views/questionnaire.py

```diff
@@ -684,17 +670,9 @@ def get_page_title_for_location(schema, current_location, context):
     return safe_content(page_title)
 
 
-def _build_template(current_location, context, template, schema):
-    previous_location = path_finder.get_previous_location(current_location, schema)
-    previous_url = previous_location.url() if previous_location is not None else None
-
-    return _render_template(context, current_location, template, previous_url, schema)
-
-
 @with_session_timeout
-@with_analytics
 @with_legal_basis
-def _render_template(
+def _build_template(
```

Comment:
The `with_legal_basis` decorator is only used on this method. It think we should remove it and add the legal basis within this method.

Reply:
Done